app/src/oculusvr/cpp/DeviceDelegateOculusVR.cpp

```diff
@@ -489,20 +506,33 @@ struct DeviceDelegateOculusVR::State {
         trackpadX = controllerState.inputState.TrackpadPosition.x / (float)controllerState.capabilities.TrackpadMaxX;
         trackpadY = controllerState.inputState.TrackpadPosition.y / (float)controllerState.capabilities.TrackpadMaxY;
 
+        controller->SetButtonState(controllerState.index, ControllerDelegate::BUTTON_TOUCHPAD,
+                (int32_t)ImmersiveButton::Touchpad, trackpadPressed, trackpadTouched);
         if (trackpadTouched && !trackpadPressed) {
           controller->SetTouchPosition(controllerState.index, trackpadX, trackpadY);
         } else {
           controller->SetTouchPosition(controllerState.index, trackpadX, trackpadY);
           controller->EndTouch(controllerState.index);
         }
-        axes[0] = trackpadTouched ? trackpadX * 2.0f - 1.0f : 0.0f;
-        axes[1] = trackpadTouched ? trackpadY * 2.0f - 1.0f : 0.0f;
+        const int32_t kNumAxes = 4;
+        float axes[kNumAxes];
+        axes[(uint32_t)ImmersiveAxis::TouchpadX] = trackpadTouched ? trackpadX * 2.0f - 1.0f : 0.0f;
+        axes[(uint32_t)ImmersiveAxis::TouchpadY] = trackpadTouched ? trackpadY * 2.0f - 1.0f : 0.0f;
```

Comment:
If touchpad x and y are 0 and 1 then we should be zeroing out 2 and 3 other wise the values will be garbage.